drivers/usb/device/usb_dc_usbip.c

```diff
@@ -0,0 +1,922 @@
+/*
+ * Copyright (c) 2018 Intel Corporation.
+ * Copyright (c) 2021 Nordic Semiconductor ASA
+ *
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+#include <string.h>
+#include <zephyr.h>
+#include <kernel.h>
+#include <sys/byteorder.h>
+#include <usb/usb_common.h>
+#include <usb/usbstruct.h>
+#include <usb/usb_device.h>
+#include <drivers/usb/usb_dc.h>
+#include <net/socket.h>
+
+#include "usb_dc_usbip.h"
+
+#include <logging/log.h>
+LOG_MODULE_REGISTER(usbip, CONFIG_USB_DRIVER_LOG_LEVEL);
+
+extern struct usb_desc_header __usb_descriptor_start[];
+
+#define USBIP_IN_EP_NUM		8
+#define USBIP_OUT_EP_NUM	8
+#define USBIP_MAX_PACKET_SIZE	64
+
+K_KERNEL_STACK_MEMBER(usbip_thread_stack, 2048);
```

Comment:
The stack size should be possible to set in Kconfig (so that user can tweak it if needed)